### DISCUSSION TITLE
@ResponseBody(or ResponseEntity<T>)가 있을 때와 없을 때의 동작 방식 차이

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#194] [Spring MVC]
.

**Comment by nayonsoso** [comment]
`@ResponseBody`(or `ResponseEntity<T>`)가 

있으면, 스프링은 컨트롤러에서 반환된 값을 HTTP 응답 본문에 직접 쓴다. 
이때 자바 객체를 자동으로 JSON 이나 XML 등의 타입으로 직렬화한다.

없으면, 스프링은 반환값을 ‘뷰 이름’으로 해석한다. 
그리고 뷰 리졸버를 사용해 뷰를 찾고, 응답한다. 
이때 뷰에 전달할 Model이 있다면, 이를 뷰에 전달한 후 응답한다.

**Comment by gyungchan-jo** [comment]
`@ResponseBody`(or ResponseEntity<T>)는 컨트롤러 메소드가 반환하는 객체를 JSON, XML 등으로 변환하여 HTTP 응답 본문에 포함시킵니다. 보통 Jackson 라이브러리를 사용하여 JSON으로 변환합니다.

만약 없다면, 반환 값은 뷰 이름으로 간주됩니다. Spring MVC는 해당 뷰 이름에 매핑되는 파일을 찾아 렌더링합니다.
